diff -aurN nmap-6.40.orig/ncat/Makefile.in nmap-6.40/ncat/Makefile.in
--- nmap-6.40.orig/ncat/Makefile.in	2013-07-30 13:13:10.132957726 +0200
+++ nmap-6.40/ncat/Makefile.in	2013-07-30 13:19:55.374499417 +0200
@@ -85,7 +85,7 @@
 ifneq ($(HAVE_LUA),)
 LUA_SRCS = ncat_lua.c
 LUA_OBJS = ncat_lua.o
-LUA_LIBS = @LIBLUA_LIBS@ -lm
+LUA_LIBS = @LIBLUA_LIBS@ -lm -ldl
 LUA_CFLAGS += -DHAVE_LUA=1 -I../liblua
 else
 LUA_SRCS =
@@ -114,7 +114,7 @@
 
 
 $(TARGET): @LUA_DEPENDS@ $(OBJS) $(NSOCKLIB)
-	$(CC) -o $@ $(CFLAGS) $(lua_cflags) $(LDFLAGS) $(OBJS) $(NSOCKLIB) $(NBASELIB) $(OPENSSL_LIBS) $(PCAP_LIBS) $(LUA_LIBS) $(LIBS)
+	$(CC) -o $@ $(CFLAGS) $(lua_cflags) $(LDFLAGS) $(OBJS) $(NSOCKLIB) $(NBASELIB) $(LUA_LIBS) $(OPENSSL_LIBS) $(PCAP_LIBS) $(LIBS)
 
 %.o: %.c
 	$(CC) $(CPPFLAGS) $(LUA_CFLAGS) $(CFLAGS) -c $< -o $@
@@ -136,16 +136,16 @@
 	fi
 
 test/addrset: test/addrset.o ncat_core.o sys_wrap.o util.o $(LUA_OBJS)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(OPENSSL_LIBS) $(PCAP_LIBS) $(LUA_LIBS)
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(LUA_LIBS) $(OPENSSL_LIBS) $(PCAP_LIBS)
 
 test/test-uri: test/test-uri.o base64.o http.o ncat_core.o  sys_wrap.o util.o $(LUA_OBJS)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(OPENSSL_LIBS) $(PCAP_LIBS) $(LUA_LIBS)
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(LUA_LIBS) $(OPENSSL_LIBS) $(PCAP_LIBS)
 
 test/test-cmdline-split: test/test-cmdline-split.o ncat_posix.o ncat_core.o sys_wrap.o util.o $(LUA_OBJS)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(OPENSSL_LIBS) $(PCAP_LIBS) $(LUA_LIBS)
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(LUA_LIBS) $(OPENSSL_LIBS) $(PCAP_LIBS)
 
 test/test-wildcard: test/test-wildcard.o ncat_core.o ncat_ssl.o sys_wrap.o util.o $(LUA_OBJS)
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(OPENSSL_LIBS) $(PCAP_LIBS) $(LUA_LIBS)
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $^ $(LIBS) $(NSOCKLIB) $(NBASELIB) $(LUA_LIBS) $(OPENSSL_LIBS) $(PCAP_LIBS)
 
 .PHONY: uninstall all clean distclean
 
